AmplifyPlugins/DataStore/Tests/AWSDataStorePluginTests/Core/AWSDataStorePluginTests.swift

```diff
@@ -90,6 +90,75 @@ class AWSDataStorePluginTests: XCTestCase {
         }
         await waitForExpectations(timeout: 1.0)
     }
+    
+    func testStorageEngineStartsOnPluginStopStart() async throws {
+        let stopExpectation = asyncExpectation(description: "Stop plugin should be called")
+        let startExpectation = expectation(description: "Start Sync should be called")
+        var currCount = 0
+        let storageEngine = MockStorageEngineBehavior()
+        
+        storageEngine.responders[.startSync] = StartSyncResponder { _ in
+            currCount = self.expect(startExpectation, currCount, 1)
+        }
+        
+        let storageEngineBehaviorFactory: StorageEngineBehaviorFactory = {_, _, _, _, _, _  throws in
+            return storageEngine
+        }
+        let dataStorePublisher = DataStorePublisher()
+        let plugin = AWSDataStorePlugin(modelRegistration: TestModelRegistration(),
+                                        storageEngineBehaviorFactory: storageEngineBehaviorFactory,
+                                        dataStorePublisher: dataStorePublisher,
+                                        validAPIPluginKey: "MockAPICategoryPlugin",
+                                        validAuthPluginKey: "MockAuthCategoryPlugin")
+        
+        do {
+            try plugin.configure(using: nil)
+            XCTAssertNil(plugin.storageEngine)
+            try await plugin.stop()
+            await stopExpectation.fulfill()
+            
+            plugin.start(completion: { _ in })
```

Comment:
the `main` PR should be identical to the `v1` PR if possible, instead of calling the async await API then the callback API, we should do exactly what we are doing in this PR over here: https://github.com/aws-amplify/amplify-swift/pull/2517/files#r1011845959

```swift
plugin.stop(completion: { _ in
                plugin.start(completion: {_ in})
            })
```

Reply:
That's definitely true for the integration tests where the CX calls into the APIs from `Amplify.DataStore`, but these tests are currently mixing both async and callback so it's not testing the bug fix or the async APIs fully